src/toxcore/binary_io.rs

```diff
@@ -74,7 +74,7 @@ impl ToBytes for Ipv6Addr {
 
 /// Parser that returns the length of the remaining input.
 pub fn rest_len(input: &[u8]) -> IResult<&[u8], usize> {
-    IResult::Done(input, input.len())
+    Ok((input, input.len()))
```

Comment:
There is rest_len in nom which we can use instead of this custom implementation.